### PR TITLE
Updated lens sprinkler

### PR DESCRIPTION
**Pull request by jbkalmbach**
I've updated the gravitational lens sprinkler in a few ways:
1) Use new functionality added to catsim by @danielsf that helps when calling multiple times to the same table in the same database (useful here for calling bulge, disk and agn at the same time). Also just updated to new catsim api since this was a little out of date already.

2) Added in the lens galaxy by replacing the row in the catalog that we choose to be our lens with new lens galaxy properties from OM10.

3) Added information to the setup page to get up and running on the sprinkler. Note this will not quite work until lsst_sims -t sims gets updated with the newest versions of sims_catUtils, sims_catalogs_generation and sims_catalogs_measures since Scott's new api functionality isn't up there yet, it's only on the github development versions at the moment.

In summary, we now have the information for the lenses and the lensed objects in a phosim catalog that we can use to create images, but something to look into is that I did notice OM10 supplies 0.0 as effective radius for the lenses.

### CODE REVIEW

**Comment by danielsf** [comment]
I know this isn't a part of the current pull request, but:

should the choice of lenses from OM10 really be random?  Put another way: if you revisit the same filed at a different MJD, do you want to be able to see the same lensing system (or am I thinking too far down the line, here)?

**Comment by danielsf** [comment]
the new lsst_sims should be published on Monday (fyi)

**Comment by drphilmarshall** [comment]
It *must* be the same lenses that we see in any given field at any MJD.
We're trying to simulate a sky survey! :-)

**Comment by danielsf** [comment]
I believe the way that @rbiswas4 dealt with this problem in the simulated supernovae catalogs is:

Every time you come to a new galaxy, re-seed numpy.random using that galaxy's uniqueId (or some other integer that CatSim guarantees will be unique and stable for all objects).  Then, draw one integer from numpy.random and use that integer to determine which of your candidate lensing systems to place in front of that galaxy.

It's not strictly random, since you are constantly re-seeding numpy.random, but it was the best we could come up with in a world where there are several billiion galaxies stored in the CatSim databases.

Rahul: do you have any idea whether that system was good enough (or have I completely mischaracterized what you ended up doing)?

**Comment by jbkalmbach** [comment]
Sorry, I didn't respond to this sooner, but since we will be talking about Twinkles next week I wanted to clear the way to get this merged. Separated the random number issue off into a separate issue and added a hard code for now concerning OM10 values that are not being calculated.

**Comment by jbkalmbach** [comment]
@SimonKrughoff This is the code I have worked on previously. What should we do about this and the new code you just merged?

**Comment by drphilmarshall** [comment]
@jbkalmbach - another delayed PR, sorry. Do you want to attempt to merge Simon's sprinkler code into your fork and report back? Ill be quicker on the merge this time, I promise..

**Comment by SimonKrughoff** [comment]
Sorry.  Do I need to do something about this?  I just noticed that I was mentioned here.

**Comment by drphilmarshall** [comment]
Nope, I just passed it to Bryce. You both edited the sprinkler, but you
merged it first so Bryce has to deal, I'm afraid. It's my fault though, he
was waiting for me to process his PR. Lesson is: don't allow Phil to be the
bottleneck for your development project.

**Comment by jchiang87** [comment]
@jbkalmbach - Could you do the merge so that Phil can process this PR and we can get your work on this into this repo?  I'd like to use your sprinkler and generatePhosimInput.py updates to create instance catalogs for the phosim benchmarking.  Thanks!

**Comment by jbkalmbach** [comment]
@jchiang87 Yes, sorry. I need to update a little of the code to make use of what Simon has added, but I will get to work on that and let you know when it's merged.

**Comment by drphilmarshall** [comment]
Thanks Bryce! :-)

**Comment by jchiang87** [comment]
@jbkalmbach Thanks for doing the merge! Your update of `generatePhosimInput.py` works out of the box, but there may be some occasions where we don't want to use the sprinkler.  If I replace
[line 84](https://github.com/DarkEnergyScienceCollaboration/Twinkles/blob/master/twinkles/generatePhosimInput.py#L84) with
```
totalCat = CompoundInstanceCatalog(compoundICList, obs_metadata=obs_metadata)
```
I get the following error:
```
Traceback (most recent call last):
  File "generatePhosimInput.py", line 99, in <module>
    generatePhosimInput()
  File "generatePhosimInput.py", line 95, in generatePhosimInput
    totalCat.write_catalog(filename)
  File "/nfs/farm/g/desc/u1/LSST_Stack/lsstsw/stack/Linux64/sims_catalogs_measures/2.1.0-2-gdca93af+2/python/lsst/sims/catalogs/measures/instance/CompoundInstanceCatalog.py", line 171, in write_catalog
    write_mode=write_mode)
  File "/nfs/farm/g/desc/u1/LSST_Stack/lsstsw/stack/Linux64/sims_catalogs_measures/2.1.0-2-gdca93af+2/python/lsst/sims/catalogs/measures/instance/CompoundInstanceCatalog.py", line 219, in _write_compound
    chunk_size=chunk_size)
  File "/nfs/farm/g/desc/u1/LSST_Stack/lsstsw/stack/Linux64/sims_catalogs_generation/2.1.0-4-ge1912d2+2/python/lsst/sims/catalogs/generation/db/dbConnection.py", line 621, in query_columns
    query = self._get_column_query(colnames)
  File "/nfs/farm/g/desc/u1/LSST_Stack/lsstsw/stack/Linux64/sims_catalogs_generation/2.1.0-4-ge1912d2+2/python/lsst/sims/catalogs/generation/db/dbConnection.py", line 533, in _get_column_query
    query = self.session.query(self.table.c[idColName].label(idLabel))
  File "/nfs/farm/g/desc/u1/LSST_Stack/lsstsw/miniconda/lib/python2.7/site-packages/sqlalchemy/util/_collections.py", line 193, in __getitem__
    return self._data[key]
KeyError: 'galtileid'
```
Is it possible to use `CompoundInstanceCatalog` without specifying a `compoundDBclass`?

**Comment by danielsf** [comment]
@jchiang87 in general, it should be possible, but galaxies are special because galtileid is generated by SQL on the fly.  Replace the sprinkler class with GalaxyTileCompoundObj from lsst.sims.catUtils.baseCatalogModels.  That ought to work.

**Comment by jbkalmbach** [comment]
Yes, since the sprinklerCompound class is just a subclass of GalaxyTileCompoundObj anyway, I don't think that will be a problem.

**Comment by jchiang87** [comment]
Thanks, guys! Using GalaxyTileCompoundObj works fine.
